agent/agent_thread.py

```diff
@@ -0,0 +1,21 @@
+import threading
+
+class AgentThread(threading.Thread):
+    def __init__(self, agent, board, player, on_find_move):
+        threading.Thread.__init__(self)
+        self.stopped = threading.Event()
+        self.daemon = True
+
+        self.on_find_move = on_find_move
+        self.agent = agent
+        self.board = board
+        self.player = player
+
+    def stop(self):
+        self.stopped.set()
+        self.join()
+
+    def run(self):
+        from ui import FPS
```

Comment:
does this not stack imports?

Reply:
`run()` can only ever be called once per a thread, so it should be okay.

Reply:
I will update it anyways, I was having circular import issues, but I figured it out.